bin/thoth

```diff
@@ -1,45 +1,43 @@
-#!/usr/bin/env node
+#!/usr/bin/env node --abort-on-uncaught-exception
```

Comment:
I've never done much inside this script, and I mostly don't understand it so I would feel better with someone other than me reviewing it as well.

Reply:
I tried and failed to do similar for the `mbucket` command. From the commit that removed that (https://github.com/joyent/node-manta/commit/27026762a9afe369fc6992f40ea77fdae7475d3f):

```
commit 27026762a9afe369fc6992f40ea77fdae7475d3f
Author: Trent Mick <trentm@gmail.com>
Date:   2019-11-15T11:12:40-08:00 (4 months ago)

    drop extra args on mbucket shebang line because that doesn't work linux

    On Linux you get:

        $ ./bin/mbucket
        /usr/bin/env: 'node --abort-on-uncaught-exception': No such file or directory
        /usr/bin/env: use -[v]S to pass options in shebang lines

    Using '-S ...' would work on this Linux and on macOS, but not on
    SmartOS. Having the --abort-on-... isn't worth it for too much
    complexity.

diff --git a/bin/mbucket b/bin/mbucket
index 3f348a0..8aab60d 100755
--- a/bin/mbucket
+++ b/bin/mbucket
@@ -1,4 +1,4 @@
-#!/usr/bin/env node --abort-on-uncaught-exception
+#!/usr/bin/env node
 /*
  * Copyright 2019 Joyent, Inc.
  */
```


To support running `bin/thoth` on Linux (or at least with whatever /usr/bin/env implementation that caused the above failure), I think we'd need to drop this.

Reply:
I don't really follow why we'd want to run thoth on Linux.

Reply:
Dunno about Linux, but I often used `thoth ls ...` commands on my Mac, so I don't think the thoth CLI is SmartOS-specific at all.

Reply:
If you find that useful, I don't intend to actively break it, though it's massively less useful without jobs for obvious reasons.